core/types.go

```diff
@@ -264,10 +264,30 @@ func RenewalInfoSimple(issued time.Time, expires time.Time) *RenewalInfo {
 	validity := expires.Add(time.Second).Sub(issued)
 	renewalOffset := validity / time.Duration(3)
 	idealRenewal := expires.Add(-renewalOffset)
+	windowStart := idealRenewal.Add(-24 * time.Hour)
+	windowEnd := idealRenewal.Add(24 * time.Hour)
+	now := time.Now().In(time.UTC)
+
+    if expires.Before(now) {
+        // If the cert has expired, return `RenewalInfoImmediate` instead.
+        return RenewalInfoImmediate(now)
+    }
+
+    if expires.Before(windowEnd) {
+        // If the windowEnd is after the cert's expiry, adjust it to the window end.
+        windowStart = expires.Add(-1 * time.Second)
+        windowEnd = expires
+    }
+    
+    if !windowEnd.After(windowStart) {
+        // Ensure windowStart is before windowEnd
+        windowEnd = expires.Add(-1 * time.Second)
+    }
```

Comment:
There is a lint error here: these lines all have incorrect indentation.

More importantly, I think there's a much simpler version of this logic that accomplishes the same goal:
- If windowEnd is after expires, make it equal to expires
- If windowStart is before issued, make it equal to issued

True, this doesn't explicitly protect against a certificate which has a zero-second validity period, but I'm not worried about that case.

Reply:
I did the first element  and changed initial spacing.  I don't know why my editor was auto-expanding tabs here.  I added the windowStart change to issued as well.

I left the last element, however, ensuring `windowStart` is BEFORE `windowEnd` - and included a reference to the RFC.  The RFC draft explicitly requires `windowStart` to precede `windowEnd`, and the payload is invalid otherwise.

Striving to simplify logic has caveats.  I prioritized ensuring an ARI compliant payload over edge cases of future notBefore values or zero second validity.